### PR TITLE
Add codegen config for customproviders, datashare, storagecache, storagesync, logic, portal

### PR DESCRIPTION
**Pull request by RodgeFu**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by RodgeFu** [comment]
Assign to codegen team for review for pure codegen config change. Thanks.

File: specification/datashare/resource-manager/readme.python.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: Microsoft.DataShare
+  package-name: datashare
+  package-version: 2019-11-01
+  clear-output-folder: true
+```
+
+```yaml $(python)
+python:
+  no-namespace-folders: true
+  output-folder: $(python-sdks-folder)/azure-mgmt/datashare
```
Comment (qiaozha) [inline specification/datashare/resource-manager/readme.python.md]: should the `azure-mgmt` suppose to be `azure-mgmt-datashare` ?

File: specification/logic/resource-manager/readme.python.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: Microsoft.Logic
+  package-name: logic
+  package-version: 2019-05-01
+  clear-output-folder: true
+```
+
+```yaml $(python)
+python:
+  no-namespace-folders: true
+  output-folder: $(python-sdks-folder)/azure-mgmt/logic
```
Comment (qiaozha) [inline specification/logic/resource-manager/readme.python.md]: here the same

File: specification/datashare/resource-manager/readme.python.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+These settings apply only when `--python` is specified on the command line.
+Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.
+
+```yaml $(python)
+python:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Microsoft.DataShare
+  package-name: datashare
```
Comment (qiaozha) [inline specification/datashare/resource-manager/readme.python.md]: should az and python have the same package name?

**Comment by NullMDR** [comment]
@qiaozha @RodgeFu Just want to confirm we should merge this PR after cli-extension onboarded?

**Comment by RodgeFu** [comment]
> @qiaozha Qiaoqiao Zhang FTE @RodgeFu Rodge Fu FTE Just want to confirm we should merge this PR after cli-extension for SDK Automation onboarded?

no, this is independent with the pipeline onboard. These config will also be used when people run codegen tool directly. thanks.

**Comment by RodgeFu** [comment]
@qiaozha , updated per comments. Please help to have a look. thanks.

File: specification/datashare/resource-manager/readme.python.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+## Python
+
+These settings apply only when `--python` is specified on the command line.
+Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.
+
+```yaml $(python)
+python:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Microsoft.DataShare
```
Comment (qiaozha) [inline specification/datashare/resource-manager/readme.python.md]: this namespace should be consistent with readme.az.md in the format of azure.mgmt.service

File: specification/logic/resource-manager/readme.python.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+## Python
+
+These settings apply only when `--python` is specified on the command line.
+Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.
+
+```yaml $(python)
+python:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Microsoft.Logic
```
Comment (qiaozha) [inline specification/logic/resource-manager/readme.python.md]: Here should also change into azure.mgmt.service

File: specification/storagesync/resource-manager/readme.python.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+## Python
+
+These settings apply only when `--python` is specified on the command line.
+Please also specify `--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`.
+
+```yaml $(python)
+python:
+  azure-arm: true
+  license-header: MICROSOFT_MIT_NO_VERSION
+  namespace: Microsoft.StorageSync
```
Comment (qiaozha) [inline specification/storagesync/resource-manager/readme.python.md]: likewise

**Review by qiaozha — APPROVED** [review]

**Comment by qiaozha** [comment]
@NullMDR  Has cli-extension automation onboard yet?

**Comment by NullMDR** [comment]
@qiaozha Not yet. You can preview generation in https://github.com/openapi-env-test/azure-rest-api-specs/pulls

**Pull request merged**